### PR TITLE
Fix checking push permission without skipping tls on configured (#628)

### PR DESCRIPTION
**Pull request by zhleonix**
When checking push permission, the skip-tls-verify options are not used, which causes failed operation on our private registry without importing root cert into the kaniko image.:
error checking push permissions -- make sure you entered the correct tag name, and that you are authenticated correctly, and try again: checking push permission for "private.repo/test/kaniko_demo:latest": Get https://private.repo/v2/: x509: certificate signed by unknown authority

A fix is required to add skipTlsVerify into httpTransport used in CheckPushPermissions.

### CODE REVIEW

**Comment by dbolshak** [comment]
@ImJasonH  Could you please review this #629 ?

Also looks like there are too many options to manage TLS issues, is there any guide about how to do this properly?

File: pkg/executor/push.go [hunk 1]
```diff
@@ -63,7 +63,15 @@ func CheckPushPermissions(opts *config.KanikoOptions) error {
 		if checked[destRef.Context().RepositoryStr()] {
 			continue
 		}
-		if err := remote.CheckPushPermission(destRef, creds.GetKeychain(), http.DefaultTransport); err != nil {
+
+		// Create a transport to set our user-agent.
+		tr := http.DefaultTransport
```
Comment (ImJasonH) [inline pkg/executor/push.go]: Should there be (is there already?) some kind of `GetTransport()` method that does this in a shared location, sort of like how `creds.GetKeychain()` does? I'm not familiar with all of the available `opts`, and even if I were, if a new opt was added that configured HTTP requests like this, it'd be nice to have that applied in one common location.

I don't think it should necessarily block this PR, but it'd be nice to get feedback from someone more familiar with kaniko.

**Comment by abergmeier** [comment]
IMO it would be better to extract this into a function. Did that in #663.

**Comment by tejal29** [comment]
@zhleonix would like to rebase this ?

**Pull request closed without merging**